ngraph/frontend/onnx/frontend/src/core/graph.cpp

```diff
@@ -289,30 +288,37 @@ OutputVector Graph::make_ng_nodes(const Node& onnx_node) const {
         NGRAPH_ERR << msg_prefix + "Unhandled exception type. \n";
         std::rethrow_exception(std::current_exception());
     }
-    set_friendly_names(onnx_node, ng_node_vector);
-    detail::add_provenance_tags(onnx_node, ng_node_vector);
+    set_friendly_names(onnx_node, ng_subgraph_outputs);
+    detail::add_provenance_tags(onnx_node, ng_subgraph_outputs);
 
     for (std::size_t i{0}; i < onnx_node.get_outputs_size(); ++i) {
-        auto ng_node = ng_node_vector.at(i);
-        m_cache->emplace_node(onnx_node.output(i), std::move(ng_node));
+        auto ng_node_output = ng_subgraph_outputs.at(i);
+        m_cache->emplace_node(onnx_node.output(i), std::move(ng_node_output));
     }
 
-    return ng_node_vector;
+    return ng_subgraph_outputs;
 }
 
-void Graph::set_friendly_names(const Node& onnx_node, const OutputVector& ng_node_vector) const {
-    for (size_t i = 0; i < ng_node_vector.size(); ++i) {
+void Graph::set_friendly_names(const Node& onnx_node, const OutputVector& ng_subgraph_outputs) const {
+    for (size_t i = 0; i < ng_subgraph_outputs.size(); ++i) {
         // Trailing optional outputs may not be specified in the ONNX model.
         // Other optional outputs should have name set to an empty string.
         if (i >= onnx_node.get_outputs_size()) {
             break;
         }
 
-        ng_node_vector[i].get_node()->set_friendly_name(onnx_node.output(i));
+        const auto& onnx_node_name = onnx_node.get_name();
+        if (onnx_node_name.empty()) {
+            // for multioutput nodes, their friendly name is always set to the last ONNX output's name
+            // this is because this setter is called in a loop and the last call is ultimate for a given node
+            ng_subgraph_outputs[i].get_node()->set_friendly_name(onnx_node.output(i));
```

Comment:
Should we come up with a common approach for all frontends? Is there any preference which output's name should be used to set a friendly name if a node itself in the model has no name? @ilyachur

Reply:
Maybe it is worth to precise that we should use **a name of first output** as a friendly name for node?

Reply:
As I know at the current moment MO tries to use tensor name for output if it is possible. In other case we have the next rule: `tensor_name/portID`
@jane-intel @popovaan Please correct me.
I propose to align with MO.

Reply:
@jane-intel @popovaan is it aligned with MO?

Reply:
@jane-intel Please take a look.

Reply:
Aligned with MO by adding `"/sink_port_" + port_index` to the current string used to set Result nodes friendly name.